### PR TITLE
Bug 1602587 - add providerMetadata to static and standalone providers

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: provider/standalone/standalone.go [hunk 1]
```diff
@@ -38,14 +38,20 @@ func (p *StandaloneProvider) ConfigureRun(state *run.State) error {
...
 
-	state.ProviderMetadata = map[string]string{}
-
 	if workerLocation, ok := p.runnercfg.Provider.Data["workerLocation"]; ok {
 		for k, v := range workerLocation.(map[string]string) {
 			state.WorkerLocation[k] = v
 		}
 	}
 
+	state.ProviderMetadata = map[string]string{}
```
Comment (petemoore) [inline provider/standalone/standalone.go]: I think this should be `map[string]interface{}{}` for properties that are `int` or `bool` etc.

File: provider/standalone/standalone.go [hunk 2]
```diff
@@ -38,14 +38,20 @@ func (p *StandaloneProvider) ConfigureRun(state *run.State) error {
...
+
+	if providerMetadata, ok := p.runnercfg.Provider.Data["providerMetadata"]; ok {
+		for k, v := range providerMetadata.(map[string]string) {
```
Comment (petemoore) [inline provider/standalone/standalone.go]: here too

File: provider/standalone/standalone_test.go [hunk 3]
```diff
@@ -8,7 +8,52 @@ import (
...
+	runnercfg := &cfg.RunnerConfig{
+		Provider: cfg.ProviderConfig{
+			ProviderType: "standalone",
+			Data: map[string]interface{}{
+				"rootURL":      "https://tc.example.com",
+				"clientID":     "testing",
+				"accessToken":  "at",
+				"workerPoolID": "w/p",
+				"workerGroup":  "wg",
+				"workerID":     "wi",
```
Comment (petemoore) [inline provider/standalone/standalone_test.go]: would be good to add a non-string here, like `"runTasksAsCurrentUser": false,`

Comment (djmitche) [reply provider/standalone/standalone_test.go]: Did you mean in the provisionerMetadata (which doesn't appear in this test)?  The set of items at Data is fixed and all of them are strings..

File: provider/standalone/standalone.go (hunk 1 above)
Comment (djmitche) [reply provider/standalone/standalone.go]: OK, I suppose people could include non-string values in the config file.

File: provider/standalone/standalone.go [hunk 4]
```diff
@@ -86,7 +97,10 @@ provider:
     workerPoolID: ..
     workerGroup: ..
     workerID: ..
-    # custom properties for TASKCLUSTER_WORKER_LOCATION
+	# (optional) custom provider-metadata entries to be passed to worker
+	providerMetadata: {prop: val, ..}
+    # (optional) custom properties for TASKCLUSTER_WORKER_LOCATION
+	# (values must be strings)
```
Comment (petemoore) [inline provider/standalone/standalone.go]: I guess we should remove this line

File: provider/static/static.go [hunk 5]
```diff
@@ -104,7 +114,10 @@ provider:
     workerGroup: ...
     workerID: ...
     staticSecret: ... # shared secret configured for this worker in worker-manager
-    # custom properties for TASKCLUSTER_WORKER_LOCATION
+	# (optional) custom provider-metadata entries to be passed to worker
+	providerMetadata: {prop: val, ..}
+    # (optional) custom properties for TASKCLUSTER_WORKER_LOCATION
+	# (values must be strings)
```
Comment (petemoore) [inline provider/static/static.go]: and this one

Comment (djmitche) [reply provider/static/static.go]: Why?  They must be strings, it'd be rude not to document that..

**Pull request merged**